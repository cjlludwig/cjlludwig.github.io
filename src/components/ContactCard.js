import headshot from '../static/images/headshot.jpg';
import '../static/components/ContactCard.css';

function Card() {
  return (
    <div className='card-wrapper'>
      <div className="card">
        <div className="top">
          <h2 className="name">Connor J Ludwig</h2>
          <img className="circle-img" src={headshot} alt="avatar_img" />
        </div>
        <div className="bottom">
          <p className="info">
            <b>Cell: </b>
            (XXX)-XXX-XXXX
          </p>
          <p className="info">
            <b>Email: </b>
            <a href = "mailto: cjlludwig@gmail.com">cjlludwig@gmail.com</a>
          </p>
          <p className="info">
            <b>Linkedin: </b>
            <a href="www.linkedin.com/in/connor-ludwig/">www.linkedin.com/in/connor-ludwig/</a>
          </p>
        </div>
      </div>
    </div>
  );
}

export default Card;