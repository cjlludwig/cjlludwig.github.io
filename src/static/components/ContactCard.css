
.card-wrapper {
  text-align: center;
  margin: 0 auto;
  padding: 25px;
  max-width: 400px;
  min-width: 400px;
}

.card {
  border-radius: 25px;
  position: relative;
  padding: 25px 15px;
  background-color: #DAD7CD;
  margin: 50px 0;
  height: 200px;
  box-shadow: 0 2px 5px #ccc;
  text-align: left;
}

.top {
  border-radius: 25px 25px 0 0;
  height: 100px;
  width: 100%;
  background-color: #A3B18A;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
}

.name {
  font-size: 1.5em;
  color: #2d3436;
  display: flex;
  flex: 1;
  margin: 10% 20px 0;
}

.card img {
  margin: 30px 20px 0 0;
}

.circle-img {
  border-radius: 50%;
  border: 7px solid #fff;
  width: 120px;
  height: 120px;
}

.bottom {
  margin-top: 90px;
}

.info {
  margin: 20px 0;
  font-size: 1em;
  color: #2d3436;
}

/* Mobile */
@media screen and (max-width:480px) {
  /* CSS for screens that are 480px pixels or less will be put in this section */
  .card-wrapper {
    text-align: center;
    margin: 0 auto;
    padding: 25px;
    max-width: 300px;
    min-width: 300px;
  }

  .name {
    font-size: 1.4em;
    color: #2d3436;
    display: flex;
    flex: 1;
    margin: 10% 20px 0;
  }
  
  .card img {
    margin: 10px 5px 0 0;
  }
  
  .circle-img {
    border-radius: 50%;
    border: 5px solid #fff;
    width: 100px;
    height: 100px;
  }

  .top {
    border-radius: 25px 25px 0 0;
    height: 70px;
    width: 100%;
    background-color: #A3B18A;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
  }

  .bottom {
    margin-top: 60px;
  }
}