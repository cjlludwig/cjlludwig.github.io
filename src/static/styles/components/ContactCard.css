
.contact-card-wrapper {
  text-align: center;
  margin: 0 auto;
  padding: 25px;
  max-width: 450px;
  min-width: 450px;
  /* width: 400px; */
}

.contact-card {
  border-radius: 25px;
  position: relative;
  padding: 25px 15px;
  background-color: #DAD7CD;
  margin: 50px 0;
  height: fit-content;
  box-shadow: 0 2px 5px #ccc;
  text-align: left;
  width: 400px;
}

.top {
  border-radius: 25px 25px 0 0;
  height: 100px;
  width: 100%;
  background-color: #A3B18A;
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
}

.name {
  font-size: 1.5em;
  color: #2d3436;
  display: flex;
  flex: 1;
  margin: 10% 20px 0;
}

.card img {
  margin: 30px 20px 0 0;
}

.circle-img {
  border-radius: 50%;
  border: 7px solid #fff;
  width: 120px;
  height: 120px;
}

.bottom {
  margin-top: 90px;
}

.info {
  margin: 10px 0;
  font-size: 1em;
  color: #2d3436;
}

/* Mobile */
@media screen and (max-width:960px) {
  /* CSS for screens that are 480px pixels or less will be put in this section */
  .contact-card-wrapper {
    text-align: center;
    margin: 0 auto;
    padding: 25px;
    max-width: 450px;
    min-width: 450px;
  }

  .name {
    font-size: 1.5em;
    color: #2d3436;
    display: flex;
    flex: 1;
    margin: 5% 20px 0;
  }
  
  .card img {
    margin: 10px 5px 0 0;
  }
  
  .circle-img {
    border-radius: 50%;
    border: 5px solid #fff;
    width: 125px;
    height: 125px;
  }

  .top {
    border-radius: 25px 25px 0 0;
    height: 70px;
    width: 100%;
    background-color: #A3B18A;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
  }

  .bottom {
    margin-top: 60px;
  }
}