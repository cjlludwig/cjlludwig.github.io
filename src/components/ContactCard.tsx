import headshot from '../static/images/cjl_headshot.png';
import '../static/styles/components/ContactCard.css';

function Card() {
  return (
    <div className='contact-card-wrapper'>
      <div className="contact-card">
        <div className="top">
          <h2 className="name">Connor J Ludwig</h2>
          <img className="circle-img" src={headshot} alt="avatar_img" />
        </div>
        <div className="bottom">
          <p className="info">
            <b>Cell: </b>
            &emsp;&emsp;(712)-830-6202
          </p>
            <b>Email: </b>
            &emsp;&ensp;<a href = "mailto: cjlludwig@gmail.com">cjlludwig@gmail.com</a>
          <p className="info">
          </p>
          <p className="info">
            <b>Github: </b>
            &emsp;<a href="https://github.com/cjlludwig">https://github.com/cjlludwig</a>
          </p>
          <p className="info">
            <b>Linkedin: </b>
            &nbsp;<a href="www.linkedin.com/in/connor-ludwig/">www.linkedin.com/in/connor-ludwig/</a>
          </p>
        </div>
      </div>
    </div>
  );
}

export default Card;