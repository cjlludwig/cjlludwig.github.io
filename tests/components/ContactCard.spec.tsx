import React from "react";
import renderer from "react-test-renderer";

import ContactCard from "../../src/components/ContactCard";

describe("<ContactCard />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<ContactCard />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});