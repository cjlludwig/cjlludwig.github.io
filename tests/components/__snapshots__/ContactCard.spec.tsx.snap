// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactCard /> renders correctly 1`] = `
<div
  className="contact-card-wrapper"
>
  <div
    className="contact-card"
  >
    <div
      className="contact-header"
    >
      <img
        alt="Connor Ludwig"
        className="profile-img"
        src=""
      />
      <div
        className="header-content"
      >
        <h2
          className="name"
        >
          Connor J Ludwig
        </h2>
        <p
          className="title"
        >
          Staff Software Engineer
        </p>
      </div>
    </div>
    <div
      className="contact-content"
    >
      <div
        className="contact-item"
      >
        <i
          className="fas fa-envelope"
        />
        <a
          href="mailto:cjlludwig@gmail.com"
        >
          cjlludwig@gmail.com
        </a>
      </div>
      <div
        className="contact-item"
      >
        <i
          className="fab fa-github"
        />
        <a
          href="https://github.com/cjlludwig"
          rel="noopener noreferrer"
          target="_blank"
        >
          github.com/cjlludwig
        </a>
      </div>
      <div
        className="contact-item"
      >
        <i
          className="fab fa-linkedin"
        />
        <a
          href="https://www.linkedin.com/in/connor-ludwig/"
          rel="noopener noreferrer"
          target="_blank"
        >
          linkedin.com/in/connor-ludwig
        </a>
      </div>
    </div>
  </div>
</div>
`;
